airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java

```diff
@@ -0,0 +1,189 @@
+ */
+
+package io.airbyte.integrations.standardtest.destination.comparator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdvancedTestDataComparator implements TestDataComparator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AdvancedTestDataComparator.class);
+
+  public static final String AIRBYTE_DATE_FORMAT = "yyyy-MM-dd";
+  public static final String AIRBYTE_DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+  public static final String AIRBYTE_DATETIME_WITH_TZ_FORMAT = "yyyy-MM-dd'T'HH:mm:ssXXX";
+
+  @Override
+  public void assertSameData(List<JsonNode> expected, List<JsonNode> actual) {
+    LOGGER.info("Expected data {}", expected);
+    LOGGER.info("Actual data   {}", actual);
+    assertEquals(expected.size(), actual.size());
+    final Iterator<JsonNode> expectedIterator = expected.iterator();
+    final Iterator<JsonNode> actualIterator = actual.iterator();
+    while (expectedIterator.hasNext() && actualIterator.hasNext()) {
+      compareObjects(expectedIterator.next(), actualIterator.next());
+    }
+  }
+
+  protected List<String> resolveIdentifier(final String identifier) {
+    return List.of(identifier);
+  }
+
+  protected void compareObjects(final JsonNode expectedObject, final JsonNode actualObject) {
+    if (!areBothEmpty(expectedObject, actualObject)) {
+      LOGGER.info("Expected Object : {}", expectedObject);
+      LOGGER.info("Actual Object   : {}", actualObject);
+      final Iterator<Map.Entry<String, JsonNode>> expectedDataIterator = expectedObject.fields();
+      assertEquals(expectedObject.size(), actualObject.size(), "Unequal row size");
+      while (expectedDataIterator.hasNext()) {
+        final Map.Entry<String, JsonNode> expectedEntry = expectedDataIterator.next();
+        final JsonNode expectedValue = expectedEntry.getValue();
+        String key = expectedEntry.getKey();
+        JsonNode actualValue = ComparatorUtils.getActualValueByExpectedKey(key, actualObject, this::resolveIdentifier);
+        LOGGER.info("For {} Expected {} vs Actual {}", key, expectedValue, actualValue);
+        assertSameValue(expectedValue, actualValue);
+      }
+    } else {
+      LOGGER.info("Both rows are empty.");
+    }
+  }
+
+  private boolean areBothEmpty(final JsonNode expectedData, final JsonNode actualData) {
+    if (expectedData.size() <= 1 && actualData.size() <= 1) {
+      boolean isExpectedEmpty = (expectedData.size() == 0 || expectedData.iterator().next().asText().isEmpty());
```

Comment:
probably it makes sense to move this specific check to a separate method and re-use it?